mir_eval/melody.py

```diff
@@ -448,63 +402,58 @@ def raw_pitch_accuracy(ref_voicing, ref_cent, est_voicing, est_cent,
     Parameters
     ----------
     ref_voicing : np.ndarray
-        Reference boolean voicing array
-    ref_cent : np.ndarray
-        Reference pitch sequence in cents
+        Reference voicing array
+    ref_freqs : np.ndarray
+        Reference pitch sequence in Hz
     est_voicing : np.ndarray
-        Estimated boolean voicing array
-    est_cent : np.ndarray
-        Estimate pitch sequence in cents
-    cent_tolerance : float
-        Maximum absolute deviation for a cent value to be considerd correct
-        (Default value = 50)
+        Estimated voicing array
+    est_freqs : np.ndarray
+        Estimate pitch sequence in Hz
+    semitone_tolerance : float
+        Maximum absolute deviation in semitones for a frequency value to be
+        considered correct
+        (Default value = 0.5)
 
     Returns
     -------
     raw_pitch : float
-        Raw pitch accuracy, the fraction of voiced frames in ref_cent for
-        which est_cent provides a correct frequency values
-        (within cent_tolerance cents).
+        Raw pitch accuracy, the fraction of voiced frames in ref_freqs for
+        which est_freqs provides a correct frequency values
+        (within semitone_tolerance).
 
     """
 
-    validate_voicing(ref_voicing, est_voicing)
-    validate(ref_voicing, ref_cent, est_voicing, est_cent)
-    ref_voicing = ref_voicing.astype(bool)
-    est_voicing = est_voicing.astype(bool)
+    validate_voicing(ref_voicing, ref_voicing)
+    validate(ref_voicing, ref_freqs, ref_voicing, est_freqs)
     # When input arrays are empty, return 0 by special case
-    if ref_voicing.size == 0 or est_voicing.size == 0 \
-       or ref_cent.size == 0 or est_cent.size == 0:
-        return 0.
     # If there are no voiced frames in reference, metric is 0
-    if ref_voicing.sum() == 0:
+    if ref_voicing.size == 0 or ref_voicing.sum() == 0:
         return 0.
 
     # Raw pitch = the number of voiced frames in the reference for which the
-    # estimate provides a correct frequency value (within cent_tolerance cents)
+    # estimate provides a correct frequency value (within semitone_tolerance)
     # NB: voicing estimation is ignored in this measure
-    matching_voicing = ref_voicing * (est_cent > 0)
-    cent_diff = np.abs(ref_cent - est_cent)[matching_voicing]
-    frame_correct = (cent_diff < cent_tolerance)
-    raw_pitch = (frame_correct).sum()/float(ref_voicing.sum())
 
-    return raw_pitch
+    nonzero_freqs = np.logical_and(est_freqs != 0, ref_freqs != 0)
+    divisor = np.abs(est_freqs[nonzero_freqs]) / ref_freqs[nonzero_freqs]
+    freq_diff_semitones = np.abs(12.0 * np.log2(divisor))
+    correct_frequencies = freq_diff_semitones < semitone_tolerance
+    rpa = np.sum(ref_voicing[nonzero_freqs] * correct_frequencies) / np.sum(ref_voicing)
+    return rpa
 
 
-def raw_chroma_accuracy(ref_voicing, ref_cent, est_voicing, est_cent,
```

Comment:
again, i'm concerned of dropping the `cent_tolerance` kwarg in a user-facing function like that without a `DeprecationWarning`